### PR TITLE
Fix cascading deserialization Spring Data's Page by setting @JsonIgnoreProperties(ignoreUnknown = true)

### PR DESCRIPTION
**Pull request by nkonev**
As discussed here https://github.com/spring-cloud/spring-cloud-openfeign/pull/636#discussion_r773598943

@KENNYSOFT @OlgaMaciaszek Please review.

### CODE REVIEW

**Comment by nkonev** [comment]
Build failure isn't related to change:
```
Could not transfer artifact org.codehaus.plexus:plexus-utils:jar:2.0.4 from/to central (https://repo.maven.apache.org/maven2): Connection reset
```

**Review by KENNYSOFT — COMMENTED** [review]
Alright, now the deserialization won't fail with unknown properties, so we may rollback `SimplePageImpl.getTotalPages()` and some methods to `@JsonProperty`, since v3.0.6 is a patch release and should not have any breaking changes. (The base branch of this PR should be 3.0.x also, to fix 3.0.6 https://github.com/spring-cloud/spring-cloud-openfeign/pull/636#pullrequestreview-817965591)

I've confirmed rollback `getTotalPages()`, `getNumberOfElements()`, `isFirst()`, `isLast()` from `@JsonIgnore` to `@JsonProperty` doesn't break any tests.

**Comment by nkonev** [comment]
Hi @OlgaMaciaszek Can you bring it to 2021.0.1 and 2020.6 releases ?

File: spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/support/PageJacksonModuleTests.java [hunk 1]
```diff
@@ -102,4 +102,57 @@ public void serializeAndDeserializeFilledMultiple() throws JsonProcessingExcepti
...
+		assertThat(cascadedResult.getTotalElements()).isEqualTo(0);
+		assertThat(cascadedResult.getContent()).hasSize(0);
+	}
+
+	@Test
+	public void serializeAndDeserializeFilledMultipleCascade() throws JsonProcessingException {
+		// Given
+		ArrayList<Object> strings0 = new ArrayList<>();
+		strings0.add("first element in cascaded serialization");
+		strings0.add("second element in cascaded serialization");
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/support/PageJacksonModuleTests.java:129]: Let's rename it to something more meaningful, like `pageElements`.

**Review by OlgaMaciaszek — CHANGES_REQUESTED** [review]
@nkonev Thanks for submitting the fix. In general looks good. However, I've added a comment about a cosmetic issue - please address it. I don't think we're going to release `3.0.x` anymore, but we will be releasing `3.1.x` - could you please submit this against `3.1.x` instead of `main` so that we can include it in both release trains.

**Comment by KENNYSOFT** [comment]
> Thanks for submitting the fix. In general looks good. However, I've added a comment about a cosmetic issue - please address it. I don't think we're going to release `3.0.x` anymore, but we will be releasing `3.1.x` - could you please submit this against `3.1.x` instead of `main` so that we can include it in both release trains.

Sad to see you're dropping support for `3.0.x` release. It's in the Spring Cloud `2020.0.x` release train but really late 2020 (December) so it was only 13 months ago.

**Review by OlgaMaciaszek — APPROVED** [review]
LGTM.

**Comment by KENNYSOFT** [comment]
@OlgaMaciaszek Why was this PR not cherry-picked to v3.0.7? 😭

**Comment by OlgaMaciaszek** [comment]
@KENNYSOFT It was my mistake. Was looking for this discussion in issues. Did not realise it was in PRs instead. Will get a `3.0.8` released just for OpenFeign later this week to get this in.

**Pull request merged**